### PR TITLE
Build: Minify bundles in parallel

### PR DESCRIPTION
**Pull request by blowery**
This adds back the ability to minify bundles in parallel. It was dropped a while back when mangling was turned on because the build would never complete with it on.

It turns out the hang was due to stderr filling up and preventing the process from exiting. This PR teaches spawn to pipe stderr to the parent, which lets it drain and close.

In addition, it uses async to limit the number of parallel processes to the number of cores available.

On an 8-way Macbook Pro, this dropped minification time by roughly half, from ~70s to ~30s.

To test, spin up the docker instance using `make docker-build && make docker-run` and verify that calypso works and bundles are minified.

### CODE REVIEW

**Comment by blowery** [comment]
Appears to cut roughly 3:20s off of a build.

File: server/bundler/bin/bundler.js [hunk 1]
```diff
@@ -42,24 +46,40 @@ function minify( files ) {
...
 			path.join( 'node_modules', '.bin', 'uglifyjs' ),
 			[
 				file,
 				'-m',
 				'-c',
 				'-o', file.replace( '.js', '.m.js' )
-			]
+			],
+			// have to pipe stderr to parent, otherwise large bundles will never finish
+			// see https://github.com/nodejs/node-v0.x-archive/issues/6764
```
Comment (samouri) [inline server/bundler/bin/bundler.js]: thanks for the link!

File: server/bundler/bin/bundler.js [hunk 2]
```diff
@@ -42,24 +46,40 @@ function minify( files ) {
...
 		}
-	} );
+		console.log( 'Minification of all bundles completed. (%dms)', Date.now() - startMinify );
+		console.log( 'Total build time: ' + ( new Date().getTime() - start ) + 'ms' );
+	}
 
-	console.log( 'Minification of all bundles completed.' );
-	console.log( 'Total build time: ' + ( new Date().getTime() - start ) + 'ms' );
+	// run one minifier per cpu core
+	async.eachLimit( files, os.cpus().length, minifyFile, done )
```
Comment (samouri) [inline server/bundler/bin/bundler.js]: very cool! I didn't realize how easy it is to create a threadpool like that

**Comment by samouri** [comment]
Code LGTM, testing it out now

**Comment by samouri** [comment]
- `docker-build`: went from 11:44 --> 9:00 (~3 mins saved as well).
- `docker-run`: I didn't notice anything funny in the console or otherwise!

**Comment by aduth** [comment]
Why is `bin/` (including this file) exempted from ESLint? There are a handful of ESLint errors in the file, including one newly introduced unused variable from this pull request (`startFile`).

**Comment by blowery** [comment]
> Why is bin/ (including this file) exempted from ESLint?

No idea, and I didn't realize! Nice catch.

**Comment by blowery** [comment]
@aduth cleaned up the eslint errors. Didn't touch the one about not using `process.exit` for now. Unsure how to duplicate that with a `throw`.

**Pull request merged**